docs/ECS Fargate Tutorials/django.md

```diff
@@ -0,0 +1,307 @@
+ aws_ecs_fargate@0.0.1    | container_definition          |            0
+ aws_ecs_fargate@0.0.1    | service_security_groups       |            0
+(17 rows)
+```
+
+## Connect to the hosted db and provision cloud resources in your AWS account
+
+1. Get a local copy of the [ECS Fargate examples repository](https://github.com/iasql/ecs-fargate-examples)
+
+```bash
+git clone git@github.com:iasql/ecs-fargate-examples.git my_project
+cd my_project
+git filter-branch --subdirectory-filter django
+```
+
+2. (Optional) Create and activate a virtual environment to install python dependencies
+
+```bash
+python -m venv <env-name>
+source <env-name>/bin/activate
+```
+
+3. Install the project dependencies under the `my_project/app` folder
+
+```bash
+cd app
+pip install -r requirements.txt
+```
+
+3. Create a `.env` file with the connection parameters provided on db creation. In this case:
+
+``` title="my_project/app/.env"
+AWS_REGION=eu-west-2
+DB_NAME=_3ba201e349a11daf
+DB_USER=qpp3pzqb
+DB_PASSWORD=LN6jnHfhRJTBD6ia
+```
+
+4. Once created, `source` the file to make the environment variables available:
+
+```bash
+source .env
+```
+
+5. (Optional) Set the desired project name that your resources will be named after by changing the `IASQL_PROJECT_NAME` in the `my_project/app/app/settings.py`. If the name is not changed, `quickstart` will be used.
+
+:::note
+
+The `project-name` can only contain alphanumeric characters and hyphens(-) because it will be used to name the load balancer
+
+:::
+
+6. Per the [Djando database documentation](https://docs.djangoproject.com/en/4.0/ref/databases/#postgresql-connection-settings-1), to connect to a new database you have to update the `DATABASES` in the `my_project/app/app/settings.py` file. This is already configure in the example project.
+
+```python title="my_project/app/app/settings.py"
+DATABASES = {
+    ...
+    'infra': {
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': environ['DB_NAME'],
+        'USER': environ['DB_USER'],
+        'PASSWORD': environ['DB_PASSWORD'],
+        'HOST': 'db.iasql.com',
+        'PORT': '5432',
+    }
+}
+```
+
+### If you are using the template example go to step 9. The following steps explains how to instrospect an existing DB in Django.
+
+7. The second migration correspond to the Django models instrospected from the modules that have been installed in the database. To introspect the schema from your database run the following command. More information [here](https://docs.djangoproject.com/en/4.0/howto/legacy-databases/).
+
+```bash
+python manage.py inspectdb --database=infra > infra/models.py
+```
+
+:::note
+
+After running the `inspectdb` command you will need to tweak the models Django generated until they work the way you’d like. 
+In our case you will have to modify the `my_project/app/infra/models.py` file as follow:
+
+1. Replace `CharField` with `TextField`
+2. Remove all `max_length=-1`
```

Comment:
Should we report this bug to the Django project?

Reply:
https://code.djangoproject.com/ticket/28612

https://code.djangoproject.com/ticket/14094

I think it has been like that forever